[minor] Automated SendGrid Subuser Management for gitops

tomklapiscak opened the pull request:
# Description

Uses the new `mas-saas-sendgrid-subuser` script from [saas-deploy-py](https://github.ibm.com/maximoappsuite/saas-deploy-py/pull/121) (now) baked into [saas-task](https://github.ibm.com/maximoappsuite/saas-task/pull/5) to automate creation/deletion of Sendgrid subusers for MAS instances when requested. This is implemented in the new `gitops-suite-config-sendgrid` and `gitops-deprovision-suite-sengrid` tasks wired into the `gitops-mas-instance` and `deprovision-mas-instance` pipelines.

This is an internal, IBM-only feature. Please refer to https://github.ibm.com/maximoappsuite/saas-tekton/pull/82 for more details and testing results.

## Associated Stories
https://jsw.ibm.com/browse/MASCORE-3422
https://jsw.ibm.com/browse/MASCORE-4623
https://jsw.ibm.com/browse/MASCORE-4805

tomklapiscak committed:
[minor] wip - first pass at gitops-suite-smtp-config-sendgrid task definition

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
wire up gitops-suite-smtp-config-sendgrid task

tomklapiscak committed:
fix

tomklapiscak committed:
add new task

tomklapiscak committed:
do not deprovision smtp config when smtp_use_sendgrid is set

tomklapiscak committed:
sendgrid subuser deprovisioning task

tomklapiscak committed:
ensure sendgrid deprovision tasks won't run (and fail) when used outside of internal IBM toolchain

tomklapiscak committed:
adding more missing params

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
prevent SMTP creds set in pipeline context from overriding values generated for sendgrid subsuser

tomklapiscak committed:
Make DNS integration optional

skip sendgrid authenticated domain setup if dns provider not specified
use mas_instance.mas_domain for default_sender_email if dns.cis.mas_domain is not specified

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
register sendgrid subuser credentials in AWS SM

tomklapiscak committed:
cleanup sendgrid_subuser AWS secret if subuser is deleted

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
fix

tomklapiscak committed:
fixes

tomklapiscak committed:
Fetch CIS API Key from AWS SM

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

whitfiea commented on tekton/src/tasks/gitops/gitops-suite-smtp-config-sendgrid.yml.j2:
```diff
@@ -0,0 +1,255 @@
+        - -c
+      name: gitops-suite-smtp-config-sendgrid
+      imagePullPolicy: Always
+      image: docker-na-public.artifactory.swg-devops.com/wiotp-docker-local/mas/saas-task:1.0.2-pre.mascore3422-amd64
```
should be latest

whitfiea commented on tekton/src/tasks/gitops/gitops-suite-smtp-config-sendgrid.yml.j2:
```diff
@@ -0,0 +1,255 @@
+        - /bin/sh
+        - -c
+      name: gitops-suite-smtp-config-sendgrid
+      imagePullPolicy: Always
```
can we change imagePullPolicy to IfNotPresent

whitfiea reviewed (dismissed):
LGTM, just needs the changes changes on the image and pull policy

tomklapiscak committed:
mark false positive secrets

tomklapiscak committed:
mark false positive secrets

tomklapiscak committed:
mark false positive secrets

tomklapiscak committed:
Gitops: Disable SendGrid Subuser (if configured) just before FVT tests run

This is to prevent the suite from sending out real emails during test execution

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
fixes

tomklapiscak committed:
Merge remote-tracking branch 'origin/master' into mascore3422

tomklapiscak committed:
saas-task:latest / IfNotPresent
